trees.c

```diff
@@ -770,12 +778,12 @@ static void send_all_trees(deflate_state *s, int lcodes, int dcodes, int blcodes
     Assert(lcodes >= 257 && dcodes >= 1 && blcodes >= 4, "not enough codes");
     Assert(lcodes <= L_CODES && dcodes <= D_CODES && blcodes <= BL_CODES, "too many codes");
     Tracev((stderr, "\nbl counts: "));
-    send_bits(s, lcodes-257, 5); /* not +255 as stated in appnote.txt */
-    send_bits(s, dcodes-1,   5);
-    send_bits(s, blcodes-4,  4); /* not -3 as stated in appnote.txt */
+    send_bits(s, lcodes-257, 5, s->bi_buf, s->bi_valid); /* not +255 as stated in appnote.txt */
+    send_bits(s, dcodes-1,   5, s->bi_buf, s->bi_valid);
+    send_bits(s, blcodes-4,  4, s->bi_buf, s->bi_valid); /* not -3 as stated in appnote.txt */
```

Comment:
True, but how often do we call send_all_trees()?
I guessed it would not matter for performance, that is why I did not use temp variables there.
If it does matter for performance, we can easily add that.

Reply:
I counted 6 indirect reads or writes per each call if not using temporary variables. Indirect reads and writes are issue on processor and compiler combinations that assume struct members are volatile by definition. 

This means the address of the pointer (relative to first member of struct) is recalculated each time and that causes slowdown. With temporary variable, only the value of the variable is modified in CPU register but not stored back in memory containing the struct contents until compiler runs out of usable registers or the current scope ends.

Reply:
This has already been addressed